testsuite/systemtest_long/qemu-openasip/tcetest_qemu-openasip.py

```diff
@@ -0,0 +1,139 @@
+DATA_PATH = os.path.join(TEST_ROOT, "data")
+MACHINE_FILE = os.path.join(DATA_PATH, "machine.adf")
+PROGRAM_COMPILE_COMMAND = "riscv64-unknown-elf-gcc -T link.ld -nostdlib -march=rv32imac -mabi=ilp32 -o crc_program.elf start.S main.c crc.c -g"
+KERNEL = os.path.join(DATA_PATH, "crc_program.elf")
+EXPECTED_RESULT = """CHECK_VALUE: 0x62488E82
+Slow CRC: 0x62488E82
+Fast CRC: 0x62488E82
+"""
+
+def qemu_available():
+    qemu_executable = os.path.join(qemu_build_dir, "qemu-system-riscv32")
+    if not os.path.isfile(qemu_executable) or not os.access(qemu_executable, os.X_OK):
+        return False
+
+    return True
+
+
+def compile_program():
+    original_dir = os.getcwd()
+    try:
+        os.chdir(DATA_PATH)        
+        result = subprocess.run(
+            shlex.split(PROGRAM_COMPILE_COMMAND),
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True
+        )
+        return True
+    except subprocess.CalledProcessError as e:
+        print(f"Compilation failed: {e}")
+        print(f"stdout: {e.stdout}")
+        print(f"stderr: {e.stderr}")
+        return False
+    finally:
+        os.chdir(original_dir)
+
+
+def run_qemu():
+    qemu_executable = os.path.join(qemu_build_dir, "qemu-system-riscv32")
+    qemu_cmd = f"{qemu_executable} " \
+               f"-machine virt,oasip_machine={MACHINE_FILE} " \
+               f"-bios none " \
+               f"-nographic " \
+               f"-no-reboot " \
+               f"-kernel {KERNEL}"
+    
+    try:
+        process = subprocess.Popen(
+            shlex.split(qemu_cmd),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True
+        )
+        output_lines = []
+        stderr_lines = []
+        found_expected_result = False
+        timeout = 10
+
+        try:
+            stdout, stderr = process.communicate(timeout=timeout)
+            output_lines = stdout.splitlines(True)
+            stderr_lines = stderr.splitlines(True)
+            found_expected_result = EXPECTED_RESULT.strip() in stdout
+        except subprocess.TimeoutExpired:
+            process.terminate()
+            stdout, stderr = process.communicate()
+            output_lines = stdout.splitlines(True)
+            stderr_lines = stderr.splitlines(True)
+            found_expected_result = EXPECTED_RESULT.strip() in stdout
+
+                
+        if not found_expected_result:
+            print("STDOUT:")
+            print(''.join(output_lines))
+            print("\nSTDERR:")
+            print(''.join(stderr_lines))
+
+        return found_expected_result
+        
+    except Exception as e:
+        print(f"QEMU execution failed: {e}")
+        try:
+            process.terminate()
+        except:
+            pass
+        return False
+    
+def run_test():
+    if not qemu_available():
+        with open(os.path.join(TEST_ROOT, "tcetest_qemu-openasip.py.disabled"), 'w') as f:
```

Comment:
This is not needed. You should have a script called 'initialize' in the directory instead which checks for qemu and touches or removes .disabled if not found or found, respectively.

Example:
```
~/src/openasip/openasip/systemtest$ cat procgen/ProGe/initialize 
#!/bin/bash
rm -rf ./proge-output
rm -rf ./proge-output-sockets
rm -fr shared-vhdl
rm -rf ./proge-output-funame
rm -rf ./proge-output-lsuport
rm -f funame_test_data.img funame_test.img funame_test.tpef
rm -f *.img *.tpef

ghdl_bin=$(which ghdl 2> /dev/null)
if [ "x${ghdl_bin}" == "x" ]; then
    touch tcetest_sram_rf.sh.disabled
else
    rm -f tcetest_sram_rf.sh.disabled
fi
```

Reply:
Also check for the availability of the needed RISC-V toolchain.